### PR TITLE
nuttx/mtd: Add new functions part_isbad and part_markbad in mtd partition.

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

need forward to the real mtd device driver

## Testing

internal test

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: drivers/mtd/mtd_partition.c [hunk 1]
```diff
@@ -856,6 +912,8 @@ FAR struct mtd_dev_s *mtd_partition(FAR struct mtd_dev_s *mtd,
   part->child.bwrite = part_bwrite;
   part->child.read   = mtd->read ? part_read : NULL;
   part->child.ioctl  = part_ioctl;
+  part->child.isbad  = part_isbad;
+  part->child.markbad  = part_markbad;
```
Comment (pkarashchenko) [inline drivers/mtd/mtd_partition.c:916]: Please align the assignment of all members of `part->child`

**Review by pkarashchenko — APPROVED** [review]

File: drivers/mtd/mtd_partition.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/mtd/mtd_partition.c]: Done.

**Comment by xiaoxiang781216** [comment]
@acassis please merge this trivial fix.

**Comment by xiaoxiang781216** [comment]
ping @acassis again

**Pull request merged**